src/building/construction_building.c

```diff
@@ -321,14 +323,43 @@ static void add_to_map(int type, building *b, int size, int orientation, int wat
     map_routing_update_walls();
 }
 
+int is_granary_cross_tile(int tile_no)
+{
+    return  tile_no == 1 ||
+        tile_no == 2 ||
+        tile_no == 3 ||
+        tile_no == 6 ||
+        tile_no == 7;
+}
+
+int is_warehouse_corner(int tile_no)
+{
+
+    int building_rot = building_rotation_get_rotation();
+    int view_rot = city_view_orientation() / 2;
+    int effective_rot = (building_rot + view_rot) % 4;
```

Comment:
It's all rot!

(Don't be afraid to use `building_rotation` and stuff, instead of `rot`.